pkg/mount/mountinfo_linux.go

```diff
@@ -90,7 +90,6 @@ func parseInfoFile(r io.Reader, filter FilterFunc) ([]*Info, error) {
 				   mount propagation flags in fields[6]. The correct
 				   behavior is to ignore any unknown optional fields.
 				*/
-				break
```

Comment:
I was wondering whether this `break` meant break from the switch or break from the for loop, but i realize i don't understand the logic in this for loop at all:

If `i` starts at 6 and it's smaller than `numFields` and `fields[i] != "-"`  only then set p.Optional, otherwise increment i, which means it will never be equal to 6 again, so p.Optional will never be set. So this entire for loop could be reduced to:
`if numFields >= 6 && fields[6] != "-" { p.Optional = fields[6] }`

Surely I must be missing something here.

Reply:
I think it's to increment `i` because that's used further down to validate if the output was correct, and to concatenation the remaining fields